python/features/alarm_clock/alarm_clock.yaml

```diff
@@ -8,40 +8,40 @@ model:
   description: A simple alarm clock
   components:
     - name: clock
-      type: Clock
+      model: Clock
     - name: timer
-      type: ClockTimer
+      model: ClockTimer
     - name: alarm
-      type: ClockAlarm
+      model: ClockAlarm
   behavior:
     - name: setAlarm
-      type: REQUEST_RESPONSE
```

Comment:
I'm not sure why it was removed, but maybe it is not necessary to define it in the core spec? I would suspect this is the case if all the example models have been reworked to leave it out, and they are still functioning. It seems that we are also no longer using the PRIMITIVE_TYPE and BEHAVIOR_TYPE constants in constants.py . Should we remove the type-related constants also?

Reply:
I removed `type` because it wasn't serving the intended purpose...and seemed to cause confusion.  Originally, the `type` field was in place for code generation purposes.  The idea was we could differentiate between things like `pub-sub` or `request-response` to generate appropriate infrastructure (i.e. synchronous vs asynchronous).  In all the examples we built, it was never really used for anything, and was often just "filled in" to satisfy the schema.  I think something similar could be useful in the future, but I think it would be better handled as a field on a plugin schema related to actual code generation.